crates/ruff/src/checkers/physical_lines.rs

```diff
@@ -139,6 +143,23 @@ pub fn check_physical_lines(
         if enforce_bidirectional_unicode {
             diagnostics.extend(pylint::rules::bidirectional_unicode(index, line));
         }
+
+        if enforce_trailing_whitespace || enforce_blank_line_contains_whitespace {
+            if let Some(diagnostic) = trailing_whitespace(
+                index,
+                line,
+                enforce_trailing_whitespace,
+                matches!(autofix, flags::Autofix::Enabled)
+                    && settings.rules.should_fix(&Rule::TrailingWhitespace),
+                enforce_blank_line_contains_whitespace,
+                matches!(autofix, flags::Autofix::Enabled)
+                    && settings
+                        .rules
+                        .should_fix(&Rule::BlankLineContainsWhitespace),
```

Comment:
For consistency, let's define these up by `fix_shebang_whitespace` using the same pattern.